common/services/github.js

```diff
@@ -364,9 +375,35 @@ module.exports = {
     const latestReleaseDate = await _getLatestReleaseDate(repoOwner, repoName);
     const now = new Date().toISOString();
     const commits = await _getCommitsWhereConfigFileHasChangedBetweenDate(repoOwner, repoName, latestReleaseDate, now);
+    const commitsShaList = commits.map((commit) => commit.sha);
     const hasConfigFileChanged = commits.length > 0;
     const latestTag = await _getLatestReleaseTagName(repoOwner, repoName);
-    return { hasConfigFileChanged, latestTag };
+
+    let pullRequests = [];
+
+    await Promise.all(
+      commitsShaList.map(async (commitSha) => {
+        const pullRequestsForCommitSha = await _getPullRequestsFromCommitSha({
+          repoOwner,
+          repoName,
+          commitSha,
+        });
+        pullRequestsForCommitSha.map((pullRequestDetails) => {
+          pullRequests.push(pullRequestDetails);
+        });
+      }),
+    );
+    const pullRequestsForCommitShaFilteredDetails = Array.from(new Set(pullRequests.map(JSON.stringify))).map(
```

Comment:
Wow, c'est technique 🤯 🦄 ⚗️

Reply:
Après c'est pour filter les doublons dans cette réponse de github, si on est sure que pour un même number 1327, c'est exactement les mêmes labels et la même html_url, on peut utiliser un reduce à la place
```
[
                {
                  number: 2438,
                  labels: [{ name: 'team-eval' }],
                  html_url: 'https://github.com/octocat/Hello-World/pull/2438',
                },
                {
                  number: 6934,
                  labels: [{ name: 'fake' }, { name: 'team-prescription' }],
                  html_url: 'https://github.com/octocat/Hello-World/pull/6934',
                },
                {
                  number: 1327,
                  labels: [{ name: 'team-captains' }, { name: 'team-acces' }],
                  html_url: 'https://github.com/octocat/Hello-World/pull/1327',
                },
              ]     
```

Reply:
En fait la doc dit qu'il vaut mieux utiliser set pour enlever des doublons.
Removing duplicate items in an array. Use [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) and [Array.from()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) instead.

La solution proposée, faire un 
`const pullRequestsForCommitShaFilteredDetails = Array.from(new Set(pullRequests));`
ne fonctionne pas